src/probe_matcher.cpp

```diff
@@ -162,7 +157,25 @@ std::set<std::string> ProbeMatcher::get_matches_for_probetype(
     }
     case ProbeType::iter:
     {
-      symbol_stream = get_iter_symbols();
+      if (bpftrace_->has_btf_data())
+      {
+        std::string ret;
+        auto iters = bpftrace_->btf_->get_all_iters();
+        for (auto& iter : iters)
+        {
+          // second check
+          if (bpftrace_->feature_->has_iter(iter))
+          {
+            ret += iter + "\n";
+          }
+          else
+          {
+            LOG(WARNING) << "The kernel broke the 'bpf_iter__' rule, please "
+                            "report this bug to us";
```

Comment:
Error message is a bit ambiguous. How about something like:

```
The kernel contains bpf_iter__$ITER struct but does not support loading an iterator program against it. Please report this bug.
```

where `$ITER` is the loop variable `iter`

Reply:
Great.